base/compiler/abstractinterpretation.jl

```diff
@@ -635,6 +635,8 @@ end
 function abstract_call(@nospecialize(f), fargs::Union{Nothing,Vector{Any}}, argtypes::Vector{Any}, vtypes::VarTable, sv::InferenceState, max_methods = sv.params.MAX_METHODS)
     if f === _apply
         return abstract_apply(argtypes[2], argtypes[3:end], vtypes, sv, max_methods)
+    elseif f === _apply_iterate
+        return abstract_apply(argtypes[3], argtypes[4:end], vtypes, sv, max_methods)
```

Comment:
```suggestion
        return abstract_apply(argtypes[2], argtypes[3], argtypes[4:end], vtypes, sv, max_methods)
```

Reply:
You're saying we should pass through the `iterate` method through abstract_apply and plumb it down into the code that does the iteration emulation? Agreed. Will do.